src/commands/generate/ckl_metadata.ts

```diff
@@ -21,23 +62,39 @@ export default class GenerateCKLMetadata extends Command {
     const {flags} = await this.parse(GenerateCKLMetadata)
     console.log('Please fill in the following fields to the best of your ability, if you do not have a value, please leave the field empty.')
     const cklMetadata = {
-      benchmark: {
-        title: prompt({ask: 'What is the benchmark title? '}) || null,
-        version: prompt({ask: 'What is the benchmark version? '}) || null,
-        plaintext: prompt({ask: 'What is the notes for release info? '}) || null,
-      },
-      stigid: prompt({ask: 'What is the STIG ID? '}) || null,
-      role: prompt({ask: 'What is the computing role? (None/Workstation/Member Server/Domain Controller) '}) || null,
-      type: _.capitalize(prompt({ask: 'What is the asset type? (Computing/Non-Computing) '})) || null,
-      hostname: prompt({ask: 'What is the asset hostname? '}) || null,
-      ip: prompt({ask: 'What is the asset IP address? '}) || null,
-      mac: prompt({ask: 'What is the asset MAC address? '}) || null,
-      tech_area: prompt({ask: 'What is the tech area? (Application Review/Boundary Security/CDS Admin Review/CDS Technical Review/Database Review/Domain Name System (DNS)/Exchange Server/Host Based System Security (HBSS)/Internal Network/Mobility/Releasable Networks (REL)/Releaseable Networks (REL)/Traditional Security/UNIX OS/VVOIP Review/Web Review/Windows OS/Other Review) '}) || null, // Yes, these typos really are how the enumerations are defined in STIG viewer's source code
-      target_key: prompt({ask: 'What is the target key? '}) || null,
-      web_or_database: prompt({ask: 'Is the target a web or database? (y/n) '}).toLowerCase() === 'y',
-      web_db_site: prompt({ask: 'What is the Web or DB site? '}) || null,
-      web_db_instance: prompt({ask: 'What is the Web or DB instance? '}) || null,
+      profiles: [
+        {
+          name: enforceNonEmptyString('What is the benchmark name? (Must match with profile name listed in HDF) '),
+          title: enforceNonEmptyString('What is the benchmark title? '),
+          version: enforceInteger('What is the benchmark version? '),
```

Comment:
test it out with the mapper proper to see what happens.  but i think version is necessary for the mapper (but it might have a default value of 0 so could be left blank here) and releasenumber and releasedate are optional fields (so could also be left blank)

Reply:
After doing some empirical testing, it is clear that there is no need to require the user to enter a version or a release number. It would take the necessary data from the provided metadata, but if not provided, it will take it from the hdf's version (formatted as a semvar). If neither are present, then 0 is used. See code [here](https://github.com/mitre/heimdall2/blob/40741b86b76df0c9f2030e6036a509c70f3b1ce3/libs/hdf-converters/src/ckl-mapper/checklist-jsonix-converter.ts#L863-L879).

The version and release number fields are now optional when generating ckl metadata.